rules.md

```diff
@@ -66,3 +66,147 @@ and fail unexpectedly when supplied extra long strings.
 ## H001: Base path can be extracted
 
 If all paths start with the same prefix then it would be cleaner to extract that into the basePath rather than repeating for each path.
+
+##
+
+
+As well as the additional rules coming from Zally, there are also additional Boat Quay rules, some of which are Backbase 
+specific but could be transferable. These additional rules, are documented below.    
+##
+
+## B012: Use Well Understood HTTP Status Codes
+
+Adapted from Zally rule 150. Well-understood codes should be used to ensure that the codes in the spec are being 
+correctly implemented. Any well-understood codes used should be used properly.
+
+### Well-understood codes:
+#### Success Codes
+Code | Method
+-----|-------
+200 | ALL
+201 | POST, PUT
+202 | POST, PUT, DELETE, PATCH
+204 | PUT, DELETE, PATCH
+207 | POST
+
+#### Redirection Codes
+Code | Method
+-----|-------
+301 | ALL
+303 | PATCH, POST, PUT, DELETE
+304 | GET, HEAD
+
+#### Client Side Error Codes
+Code | Method
+-----|-------
+400 | ALL
+401 | ALL
+403 | ALL
+404 | ALL
+405 | ALL
+406 | ALL
+408 | ALL
+409 | POST, PUT, DELETE, PATCH
+410 | ALL
+412 | PUT, DELETE, PATCH
+415 | POST, PUT, DELETE, PATCH
+422 | ALL
+423 | PUT, DELETE, PATCH
+428 | ALL
+429 | ALL
+
+#### Server Side Error Codes
+Code | Method
+-----|-------
+    500 | ALL
+    501 | ALL
+    502 | ALL
+    503 | ALL
+    504 | ALL
+
+## B011: Use standard HTTP status codes
+
+Adapted from Zally rule 150, this adds extra HTTP codes to the list of standard and well-understood codes. Any HTTP codes 
+used in responses must be from the standard list.
+Standard HTTP status codes:
+
+    100, 101, 200, 201, 202, 203, 204, 205, 206, 207
+    300, 301, 302, 303, 304, 305, 307, 400, 401, 402, 
+    403, 404, 405, 406, 407, 408, 409, 410, 411, 412, 
+    413, 414, 415, 416, 417, 422, 423, 426, 428, 429, 
+    431, 500, 501, 502, 503, 504, 505, 511, default
+    
```

Comment:
This list of 'standard codes' - are they supposed to be supported?
Because most of them are not..From this list the following ones will fail the build if failOnWarning is set to true
```
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 100 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 101 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 300 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 302 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 205 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 206 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 203 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 305 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 307 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 417 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 416 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 407 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 411 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 414 is not a well-understood response code
[WARNING] [150] SHOULD - Use Standard HTTP Status Codes: 413 is not a well-understood response code
[WARNING] [B012] SHOULD - Use Well Understood HTTP Status Codes: 423 is not allowed for this method, incorrect use of well known code
[WARNING] [B012] SHOULD - Use Well Understood HTTP Status Codes: 426 is not a well understood status code, should use well-understood HTTP status codes
[WARNING] [B012] SHOULD - Use Well Understood HTTP Status Codes: 505 is not a well understood status code, should use well-understood HTTP status codes
[WARNING] [B012] SHOULD - Use Well Understood HTTP Status Codes: 431 is not a well understood status code, should use well-understood HTTP status codes
[WARNING] [B012] SHOULD - Use Well Understood HTTP Status Codes: 511 is not a well understood status code, should use well-understood HTTP status codes
```

Reply:
Do you think the severity level should be added to every rule? Like how zalando is very clear about the difference between MUST and SHOULD at https://opensource.zalando.com/restful-api-guidelines/?